linera-service/tests/linera_net_tests.rs

```diff
@@ -624,6 +689,209 @@ async fn test_evm_end_to_end_child_subcontract(config: impl LineraNetConfig) ->
+
+    // Checking the balances on input
+
+    assert_contract_balance(&app_a, address1, balance_a_1).await?;
+    assert_contract_balance(&app_a, address2, balance_a_2).await?;
+    assert_contract_balance(&app_a, address_app, balance_a_app).await?;
+
+    // Transfering amount
+
+    let amount = Amount::from_tokens(1);
+    let operation = send_cashCall {
+        recipient: address2,
+        amount: amount.into(),
+    };
+    let operation = get_zero_operation(operation)?;
+    app_a.run_json_query(operation).await?;
+
+    // Checking the balances of app_a
+
+    let balance_a_1_after = node_service_a.balance(&account_a_1).await?;
+    let balance_a_2_after = node_service_a.balance(&account_a_2).await?;
+    let balance_a_app_after = node_service_a.balance(&account_a_app).await?;
+    assert_eq!(balance_a_1_after, balance_a_1);
+    assert_eq!(balance_a_2_after, balance_a_2 + amount);
+    assert_eq!(balance_a_app_after, balance_a_app - amount);
+
+    assert_contract_balance(&app_a, address1, balance_a_1_after).await?;
+    assert_contract_balance(&app_a, address2, balance_a_2_after).await?;
+    assert_contract_balance(&app_a, address_app, balance_a_app_after).await?;
+
+    // Doing an operation with a non-zero amount
+    let operation = null_operationCall {};
+    let amount_operation = Amount::from(2);
+    let operation = EvmOperation::new(amount_operation, operation.abi_encode());
+    let operation = EvmQuery::Operation(operation.to_bytes()?);
+    app_a.run_json_query(operation).await?;
+
+    let balance_a_app_after2 = node_service_a.balance(&account_a_app).await?;
+    assert_eq!(balance_a_app_after2, balance_a_app_after + amount_operation);
+
+    // Creating app_b via null_operation and checking balances.
+    let account_b_1 = Account {
+        chain_id: chain_b,
+        owner: account_owner1,
+    };
+    let account_b_2 = Account {
+        chain_id: chain_b,
+        owner: account_owner2,
+    };
+    let account_b_app = Account {
+        chain_id: chain_b,
+        owner: account_owner_app,
+    };
+
+    let operation = null_operationCall {};
+    let operation = get_zero_operation(operation)?;
+    app_b.run_json_query(operation).await?;
+
+    assert_eq!(node_service_b.balance(&account_b_1).await?, Amount::ZERO);
+    assert_eq!(node_service_b.balance(&account_b_2).await?, Amount::ZERO);
+    assert_eq!(node_service_b.balance(&account_b_app).await?, Amount::ZERO);
+    assert_contract_balance(&app_b, address1, Amount::ZERO).await?;
+    assert_contract_balance(&app_b, address2, Amount::ZERO).await?;
+    assert_contract_balance(&app_b, address_app, Amount::ZERO).await?;
+
+    // Creating app_b via service calls and checking balances.
+    assert_contract_balance(&app_c, address1, Amount::ZERO).await?;
+    assert_contract_balance(&app_c, address2, Amount::ZERO).await?;
+    assert_contract_balance(&app_c, address_app, Amount::ZERO).await?;
```

Comment:
Did we do anything with app C and node service C at all? Do we need three node services?

Actually, did A and B even interact? Or are these basically two independent test cases?